updater/lib/dependabot/updater/error_handler.rb

```diff
@@ -66,20 +77,36 @@ def handle_dependency_error(error:, dependency:, dependency_group: nil)
       end
 
       # Provides logging for errors that occur when processing a dependency
+      sig do
+        params(
+          dependency: T.nilable(Dependabot::Dependency),
+          error: StandardError,
+          error_type: String,
+          error_detail: T.nilable(T.any(String, T::Hash[T.untyped, T.untyped]))
```

Comment:
Can you be more specific with the hash?

Reply:
Nothing obvious breaks if I change it to `T::Hash[Symbol, T.untyped]`, [but other loggers](https://github.com/dependabot/dependabot-core/blob/7f7108ceb8edde9ddc8f5e0f9f67d0dc2e7eb9cb/updater/lib/dependabot/api_client.rb#L102) that take an `error_details` in this area take `T::Hash[T.untyped, T.untyped]` so I figured it best to leave it open rather than cause a type-error on a log method since we're just going to be doing `to_s` anyway.

I tried not to spread `T.untyped` around too much, but in this case it really seems that we don't case about what the type is, so it seemed safe.